addon/components/search-result/component.js

```diff
@@ -0,0 +1,170 @@
+            return this.get('result.description').slice(0, this.get('maxDescription'));
+        }
+    }),
+    allCreators: Ember.computed('result.lists.contributors', function() {
+        return (this.get('result.lists.contributors') || []).filter(contrib => contrib.relation === 'creator').sort(function(a, b) {
+            return a.order_cited - b.order_cited;
+        });
+    }),
+    extraCreators: Ember.computed('allCreators', function() {
+        return this.get('allCreators').slice(this.get('maxCreators'));
+    }),
+    creators: Ember.computed('allCreators', function() {
+        return this.get('allCreators').slice(0, this.get('maxCreators'));
+    }),
+    extraTags: Ember.computed('result.tags', function() {
+        return (this.get('result.tags') || []).slice(this.get('maxTags'));
+    }),
+    identifiers: Ember.computed('result.identifiers', function() {
+        return this.get('result.identifiers');
+    }),
+    tags: Ember.computed('result.tags', function() {
+        return (this.get('result.tags') || []).slice(0, this.get('maxTags'));
+    }),
+    subjects: Ember.computed('result.subjects', function() {
+        return (this.get('result.subjects') || []).slice(0, this.get('maxSubjects'));
+    }),
+    extraSubjects: Ember.computed('result.subjects', function() {
+        return (this.get('result.subjects') || []).slice(this.get('maxSubjects'));
+    }),
+    retractionId: Ember.computed('result.lists.retractions[]', function() {
+        const retractions = this.get('result.lists.retractions');
+        if (retractions && retractions.length) {
+            return retractions[0].id;
+        }
+        return null;
+    }),
+    osfID: Ember.computed('result', function() {
+        let re = /osf.io\/(\w+)\/$/;
+        // NOTE / TODO : This will have to be removed later. Currently the only "true" preprints are solely from the OSF
+        // socarxiv and the like sometimes get picked up by as part of OSF, which is technically true. This will prevent
+        // broken links to things that aren't really preprints.
+        if (this.get('result.providers.length') === 1 && this.get('result.providers').find(provider => provider.name === 'OSF'))
+            for (let i = 0; i < this.get('result.identifiers.length'); i++)
+                if (re.test(this.get('result.identifiers')[i]))
+                    return re.exec(this.get('result.identifiers')[i])[1];
+        return false;
+    }),
+    hyperlink: Ember.computed('result', function() {
+        let re = null;
+        for (let i = 0; i < this.get('result.providers.length'); i++) {
+            //If the result has multiple providers, and one of them matches, use the first one found.
+            re = this.providerUrlRegex[this.get('result.providers')[i].name];
+            if (re) break;
+        }
+
+        re = re || this.providerUrlRegex.OSF;
+
+        const identifiers = this.get('result.identifiers').filter(ident => ident.startsWith('http://'));
```

Comment:
Will identifiers ever start with `https`?

(https by default is getting increasingly more common)